Sources/Paywalls/Components/PaywallStackComponent.swift

```diff
@@ -54,6 +60,17 @@ public extension PaywallComponent {
         }
 
     }
+
 }
 
+public extension PaywallComponent {
+
+    // Components that can be contained in a stack
+    protocol StackableComponent: PaywallComponentBase {}
+
+}
+
+extension PaywallComponent: PaywallComponent.StackableComponent {}
+extension PaywallComponent.PackageComponent: PaywallComponent.StackableComponent {}
```

Comment:
For my understanding, could you explain what this is used for?

Reply:
Of course! So, I wanted to be _very_ specific in the types that could be used by `struct GenericStackComponent<T: StackableComponent>`

So I created the new `StackableComponent` component protocol and then added it to `PaywallComponent` (the enum of all the types) and `PaywallComponent.PackageComponent` (the specific package type).

I _wanted_ to do a javascript thing like `PaywallComponent  | PaywallComponent.PackageComponent` but that doesn't really exist in Swift so this is the _very_ verbose way of doing it 😛

Reply:
Oh nice! Make-your-own-union-type! 😄 In Kotlin my first try would be a sealed type, but this is an interesting approach!